lib/PipelineC/Tracing/Runner.cpp

```diff
@@ -0,0 +1,429 @@
+    // clang-format on
+    return Argument.toIntList<std::remove_pointer_t<ArgT>>();
+  }
+
+  template<ConstexprString Name, typename ArgT>
+  // clang-format off
+  requires (std::is_pointer_v<ArgT> && isRPType<remove_constptr<ArgT>>())
+  ArgStorage<ArgT> parseArgumentImpl(ArgumentRef Argument) {
+    // clang-format on
+    revng_check(Argument.isScalar(), "Argument is not scalar");
+    return getPointer<ArgT>(Argument.getScalar(), isDestroy<Name>());
+  }
+
+  template<ConstexprString Name, typename ArgT>
+  // clang-format off
+  requires (isList<ArgT>() &&
+            isRPType<remove_constptr<remove_constptr<ArgT>>>())
+  ArgStorage<ArgT> parseArgumentImpl(ArgumentRef Argument) {
+    // clang-format on
+    revng_check(Argument.isSequence(), "Argument is not a sequence");
+    using ElementT = std::remove_pointer_t<ArgT>;
+    std::vector<ElementT> Result;
+    for (auto &Elem : Argument.getSequence()) {
+      Result.push_back(getPointer<ElementT>(Elem, isDestroy<Name>()));
+    }
+
+    utils::poisonTail(Result);
+    return Result;
+  }
+
+public :
+
+  template<ConstexprString Name, typename ArgT, size_t I>
+  ArgStorage<ArgT>
+  parseArgument(ArgumentsRef Arguments) {
+    if constexpr (anyOf<ArgT, char *, const char *>())
+      return parseString<Name, ArgT, I>(Arguments[I]);
+    else
+      return parseArgumentImpl<Name, ArgT>(Arguments[I]);
+  }
+
+  template<ConstexprString Name, typename ArgT, size_t I, typename... Args>
+  ArgT unwrapStorage(std::tuple<Args...> Arguments) {
+    if constexpr (isList<ArgT>()) {
+      return (ArgT) std::get<I>(Arguments).data();
+    } else {
+      if constexpr (anyOf<ArgT, char *, const char *>()) {
+        if constexpr (isDestroy<Name>()) {
+          // _destroy method parameter is not a string, but a pointer to the
+          // string's address
+          ArgStorage<ArgT> Vector = std::get<I>(Arguments);
+          return getPointer<ArgT>(std::string(Vector.data()), true);
+        } else {
+          // Strings are stored as std::vector<char>
+          return std::get<I>(Arguments).data();
+        }
+      } else {
+        return std::get<I>(Arguments);
+      }
+    }
+  }
+};
+
+// clang-format off
+using CommandRunner = std::function<void(RunnerContext &Context,
+                                         ArgumentsRef Arguments,
+                                         ReturnRef Return)>;
+// clang-format on
+
+static std::map<std::string, CommandRunner> CommandHandler;
+
+template<ConstexprString Name, typename ReturnT, typename... Args, size_t... I>
+static ReturnT runCommand(std::function<ReturnT(Args...)> Function,
+                          ArgumentsRef Arguments,
+                          RunnerContext &Context,
+                          std::index_sequence<I...> Seq) {
+  using std::make_tuple;
+  auto &Arg = Arguments;
+  using StorageT = std::tuple<ArgStorage<Args>...>;
+  StorageT Storage = make_tuple(Context.parseArgument<Name, Args, I>(Arg)...);
```

Comment:
```suggestion
  auto &Arg = Arguments;
  using StorageT = std::tuple<ArgStorage<Args>...>;
  StorageT Storage = { Context.parseArgument<Name, Args, I>(Arg)... };
```

Reply:
It cannot be done this way. The main issue is buffers (and, in case, string arrays) as those need to be dealt with in 2 steps:
1. Read the source argument and save the "transformed" data in some buffer (e.g. `std::vector<char>`)
2. Run the `Context.parseArgument` to return the type expected by PipelineC (e.g. `const char *`)

This cannot be done in one single step because otherwise the storage would be in the `parseArgument` function and it would end at the function's `return`

Reply:
Are you sure your comment is relevant?
I simply dropped `make_tuple` in favor of `{ ... }`.

Reply:
Sorry I misread! switched to braces